plugin/plugin.go

```diff
@@ -21,11 +21,23 @@ type CodeGenerator interface {
 }
 
 // EarlySourceInjector is used to inject things that are required for user schema files to compile.
+// Deprecated: Use EarlySourcesInjector instead
 type EarlySourceInjector interface {
 	InjectSourceEarly() *ast.Source
 }
 
+// EarlySourcesInjector is used to inject things that are required for user schema files to compile.
+type EarlySourcesInjector interface {
+	InjectSourcesEarly() ([]*ast.Source, error)
```

Comment:
Instead of having empty parameter, would it be possible to pass the config? For example `InjectSourcesEarly(cfg)` in case we want to do stuff with some configurations from the file?